inline.h

```diff
@@ -3451,8 +3451,7 @@ Perl_mortal_getenv(const char * str)
 PERL_STATIC_INLINE bool
 Perl_sv_isbool(pTHX_ const SV *sv)
 {
-    return SvIOK(sv) && SvPOK(sv) && SvIsCOW_static(sv) &&
-        (SvPVX_const(sv) == PL_Yes || SvPVX_const(sv) == PL_No);
+    return SvBoolFlagsOK(sv) && BOOL_INTERNALS_sv_isbool(sv);
```

Comment:
As the rest of this PR is purely additions, I think the rest of it is fine to go in to 5.35.11 at this late hour.

However, these three lines aren't *exactly* equivalent any more because of the slightly different way the flags tests are done. I'd feel more comfortable approving it without this part to it - we can maybe add a comment though to remind ourselves to update it to using this form when we start development for 5.37.

Reply:
Sure, done. Although I have to admit I don't really understand your concern here, at the mathematical / logical level they are exactly equivalent operations if they weren't on some machine then something would be very very wrong with that machine. I suspect that an optimizing compiler will change the less efficient form into the more efficient form anyway.

FWIW I ran this:
```
PERL_STATIC_INLINE bool
Perl_sv_isbool(pTHX_ const SV *sv)
{
#ifdef DEBUGGING
    bool n = (SvBoolFlagsOK(sv) && BOOL_INTERNALS_sv_isbool(sv));
    bool o = (SvIOK(sv) && SvPOK(sv) && SvIsCOW_static(sv) &&
            (SvPVX_const(sv) == PL_Yes || SvPVX_const(sv) == PL_No));
    assert(n==o);
#endif
    return SvIOK(sv) && SvPOK(sv) && SvIsCOW_static(sv) &&
        (SvPVX_const(sv) == PL_Yes || SvPVX_const(sv) == PL_No);

}
```
and no asserts fired, as expected. Anyway, I have pushed a patch as your requested. I used an ifdef 0 to leave the new code with a comment that we will switch in 5.37.

Reply:
Oh nice - I like the `#if 0` approach to this.
Thanks.